src/Symfony/Component/VarExporter/ProxyHelper.php

```diff
@@ -197,10 +197,36 @@ public static function generateLazyProxy(?\ReflectionClass $class, array $interf
         $body = $methods ? "\n".implode("\n\n", $methods)."\n" : '';
         $propertyScopes = $class ? self::exportPropertyScopes($class->name) : '[]';
 
+        if (
+            $class?->hasMethod('__unserialize')
+            && null !== ($unserializeParameter = $class->getMethod('__unserialize')->getParameters()[0] ?? null)
+            && null === $unserializeParameter->getType()
```

Comment:
you spotted it already: the engine checks that there is one parameter so we don't need the `?? null` part ;)
```suggestion
            && $class->getMethod('__unserialize')->getParameters()[0]->getType()
```